test/test_unistd.rs

```diff
@@ -104,6 +104,57 @@ mod linux_android {
     }
 }
 
+#[test]
+#[cfg(not(any(target_os = "ios", target_os = "macos")))]
+fn test_setgroups() {
+    if !Uid::current().is_root() {
+        // setgroups() requires root
+        return
+    }
+
+    // Save the existing groups
+    let old_groups = getgroups().unwrap();
+
+    // Set some new made up groups
+    let groups = [Gid::from_raw(123), Gid::from_raw(456)];
+    setgroups(&groups).unwrap();
+
+    let new_groups = getgroups().unwrap();
+    assert_eq!(new_groups, groups);
+
+    // Revert back to the old groups
+    setgroups(&old_groups).unwrap();
+}
+
+#[test]
+#[cfg(not(any(target_os = "ios", target_os = "macos")))]
+fn test_initgroups() {
+    if !Uid::current().is_root() {
+        // initgroups(), setgroups() require root
```

Comment:
Move this comment above the `if` and rephrase to something like "Skip this test when not run as `root` as `initgroups()` and `setgroups()` both require root."